### PR TITLE
Feature multiple preset selection up down selection

### PR DESCRIPTION
**Pull request by schlenkibus**
works kinda

### CODE REVIEW

File: NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java [hunk 1]
```diff
@@ -68,4 +68,41 @@ public String getCSV() {
 	public void deletePresets() {
 		NonMaps.get().getServerProxy().deletePresets(getCSV());
 	}
+		
+	private PresetManager getPresetManager() {
+		return NonMaps.get().getNonLinearWorld().getPresetManager();
+	}
+	
+	private String getLastUuid() {
```
Comment (hhoegelo) [inline NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java]: will throw if called on empty selection

File: NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java [hunk 2]
```diff
@@ -68,4 +68,41 @@ public String getCSV() {
...
+		return selectedPresets.get(selectedPresets.size() - 1);
+	}
+		
+	public void handleDownKey() {		
+		Preset curr = getPresetManager().findPreset(getLastUuid());
+		if(curr != null) {
+			Preset next = getPresetManager().findPreset(getLastUuid()).getParent().getPresetList().getNext(getLastUuid());
+			if(next != null) {
+				if(contains(next)) {
+					remove(curr);
```
Comment (hhoegelo) [inline NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java]: why remove 'curr' but add 'next'?

File: NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java [hunk 3]
```diff
@@ -68,4 +68,41 @@ public String getCSV() {
...
+		
+	}
+	
+	public void handleUpKey() {
+		Preset curr = getPresetManager().findPreset(getLastUuid());
+		if(curr != null) {
+			Preset prev = getPresetManager().findPreset(getLastUuid()).getParent().getPresetList().getPrev(getLastUuid());
+			if(prev != null) {
+				if(contains(prev)) {
+					remove(curr);
```
Comment (hhoegelo) [inline NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java]: why remove 'curr' but add 'prev'?

File: NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java (hunk 1 above)
Comment (schlenkibus) [reply NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java]: not anymore

File: NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java (hunk 2 above)
Comment (schlenkibus) [reply NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java]: well thats just how it is

File: NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java (hunk 3 above)
Comment (schlenkibus) [reply NonMaps/src/com/nonlinearlabs/NonMaps/client/world/maps/presets/MultiplePresetSelection.java]: thats just how it is

**Pull request merged**